From e44b64d297293b83459fddda968923750e2f6a5b Mon Sep 17 00:00:00 2001
From: Eike Rathke <erack@redhat.com>
Date: Thu, 2 Feb 2012 19:11:37 +0100
Subject: [PATCH] resolved fdo#42784 BorderLine with only InnerWidth set does
 not work (cherry picked from commit
 971f826a96f0938bfcfd8dfe26edd50d8e031bb4)

Signed-off-by: Michael Stahl <mstahl@redhat.com>
---
 editeng/source/items/borderline.cxx |   16 ++++++++++++++++
 1 files changed, 16 insertions(+), 0 deletions(-)

diff --git a/editeng/source/items/borderline.cxx b/editeng/source/items/borderline.cxx
index 61cecce..d15f53c 100644
--- a/editeng/source/items/borderline.cxx
+++ b/editeng/source/items/borderline.cxx
@@ -300,6 +300,22 @@ void SvxBorderLine::GuessLinesWidths( SvxBorderStyle nStyle, sal_uInt16 nOut, sa
     else
     {
         SetStyle( nStyle );
+        if (nOut == 0 && nIn > 0)
+        {
+            // If only inner width is given swap inner and outer widths for
+            // single line styles, otherwise GuessWidth() marks this as invalid
+            // and returns a 0 width.
+            switch (nStyle)
+            {
+                case SOLID:
+                case DOTTED:
+                case DASHED:
+                    ::std::swap( nOut, nIn);
+                    break;
+                default:
+                    ;   // nothing
+            }
+        }
         m_nWidth = m_aWidthImpl.GuessWidth( nOut, nIn, nDist );
     }
 }
-- 
1.7.7.6

